app/helpers/dashboards_helper.rb

```diff
@@ -18,4 +18,37 @@ def result_type(result)
   def status_tags(result)
     StatusTagService.run(resource: result)
   end
+
+  def result_date(result)
+    return if refused_or_revoked?(result)
+    return expired_date(result) if result.metaData.EXPIRED?
+    return will_expire_date(result) if result.is_a?(WasteCarriersEngine::Registration)
+
+    last_modified_date(result)
+  end
+
+  private
+
+  def refused_or_revoked?(result)
+    %w[REFUSED REVOKED].include?(result.metaData.status)
```

Comment:
I do feel like we are giving helpers knowledge that belongs to the model in regard to accessing the status. Shall we create a `state` method in the models and use that to decide about the status? Even better if we can implement something like:
```
def revoked?
  metaData.status == "REVOKED"
end

def refused?
  metaData.status == "REFUSED"
end
```
In the engine model. 
I would love to have states managed in the same place so that/if they will need update we won't have to change things all over the place. 
In general, hoping for the killing of MongoDB in favour of Postgres, if we can keep every access to `metaData` inside the model and anything else needing it has to pass through a model public interface method, it would be good. 
Specifically, I really would like to see states managed all in one place, proper constant created for the different states and they always accessed through their owner's public interface.

Reply:
I agree with this, though it'd probably need to be a concern applied to both registration and transient_registration (you can refuse a renewal).

However I'd expect it done in a separate PR, not as part of this one. So i'm fine with holding this until that PR is done and merged, or approving this but then submitting the statuses PR's straight after.

Reply:
Feast yer eyes on https://github.com/DEFRA/waste-carriers-engine/pull/520

Reply:
And fixed: https://github.com/DEFRA/waste-carriers-back-office/pull/387/commits/4228992a86e96b0aedbcbb98757bffa521665bf6